x/upgrade/module.go

```diff
@@ -177,10 +177,11 @@ func init() {
 type ModuleInputs struct {
 	depinject.In
 
-	Config       *modulev1.Module
-	Key          *store.KVStoreKey
-	Cdc          codec.Codec
-	AddressCodec address.Codec
+	Config           *modulev1.Module
+	Key              *store.KVStoreKey
+	Cdc              codec.Codec
+	AddressCodec     address.Codec
+	CommitMultiStore store.CommitMultiStore
```

Comment:
Have no idea if it's correct, not familiar with new depinject stuff.

Reply:
This is correct, but you need something that will provide the CommitMultiStore().

Reply:
Where should I put that?

Reply:
Given that it is instantiated after `appBuilder.Build`, AFAIK you actually need to use a setter and not have this in the keeper, and do something like `PopulateVersionMap` but for the CMS.
Then you can invoke that function in the init function. I'd say something like this in x/upgrade/module.go:

```diff
	appmodule.Register(&modulev1.Module{},
		appmodule.Invoke(PopulateVersionMap),
+		appmodule.Invoke(PopulateCommitMultiStore),
	)
```

```go
func PopulateCommitMultiStore(upgradeKeeper *keeper.Keeper, bApp *baseapp.BaseApp) {
	if upgradeKeeper == nil {
		return
	}

	upgradeKeeper.SetCommitMultiStore(bApp.CommitMultiStore())
}
```

Reply:
Given that it is instantiated after `appBuilder.Build`, AFAIK you actually need to use a setter and not have this in the keeper, and do something like `PopulateVersionMap` but for the CMS.
Then you can invoke that function in the init function. I'd say something like this in x/upgrade/module.go:

```diff
	appmodule.Register(&modulev1.Module{},
		appmodule.Invoke(PopulateVersionMap),
+		appmodule.Invoke(PopulateCommitMultiStore),
	)
```

```go
func PopulateCommitMultiStore(upgradeKeeper *keeper.Keeper, bApp *baseapp.BaseApp) {
	if upgradeKeeper == nil {
		return
	}

	upgradeKeeper.SetCommitMultiStore(bApp.CommitMultiStore())
}
```

Note, that I have not tested the above.